cg/meta/demultiplex/utils.py

```diff
@@ -232,3 +235,39 @@ def get_flow_cell_id(flow_cell_dir_name: str) -> str:
     Example: 230802_A00689_0857_BHGTMFDSX7 -> HGTMFDSX7
     """
     return flow_cell_dir_name.split("_")[-1][1:]
+
+
+def needs_manifest_file(flow_cell_dir: Path) -> bool:
+    """Returns whether a flow cell directory needs a manifest file."""
+    illumina_manifest_file = Path(flow_cell_dir, DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST)
+    custom_manifest_file = Path(
+        flow_cell_dir, DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST
+    )
+    copy_complete_file = Path(flow_cell_dir, DemultiplexingDirsAndFiles.COPY_COMPLETE)
+    return (
+        not any((illumina_manifest_file.exists(), custom_manifest_file.exists()))
+        and copy_complete_file.exists()
+    )
+
+
+def create_manifest_file(flow_cell_dir_name: Path) -> Path:
+    """Creates a tab separated file containing the paths of all files in the given
+    directory and any subdirectories."""
+    files_in_directory: list[list[str]] = []
+    for subdir, _, files in os.walk(flow_cell_dir_name):
+        subdir = Path(subdir).relative_to(flow_cell_dir_name)
+        files_in_directory.extend([Path(subdir, file).as_posix()] for file in files)
```

Comment:
would make this a generic files function in utilties so it can be re-used and tested separately

```suggestion
    files_in_directory: list[list[str]] = []
    for subdir, _, files in os.walk(flow_cell_dir_name):
        subdir = Path(subdir).relative_to(flow_cell_dir_name)
        files_in_directory.extend([Path(subdir, file).as_posix()] for file in files)
```

Reply:
I'm not sure if this would be useful elsewhere. The added paths are relative to the given directory and the output format is `list[list[str]]` (due to what is used by the csv_writer). 

If you think the function would be clearer by extracting it I would be OK with that but it is pretty short as is.

Reply:
Is this not the same code that is used in the demultiplex/utils.py function? 
Here 
```Python
files_in_directory: list[list[str]] = []
    for subdir, _, files in os.walk(flow_cell_dir_name):
        subdir = Path(subdir).relative_to(flow_cell_dir_name)
        files_in_directory.extend([Path(subdir, file).as_posix()] for file in files)
```
Below
```Python
    all_files: list[Path] = []
    for subdir, _, files in os.walk(tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert):
        subdir = Path(subdir).relative_to(
            tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert
        )
        all_files.extend([Path(subdir, file) for file in files])
```
Except for the output type its the same. Maybe there is a way to combine these as `get_relative_file_paths` I think we use that in the housekeeper context sometimes so it might be useful there. Up to you

Reply:
Since there is only one usage in the production code I leave it be.